internal/k8shandler/vector.go

```diff
@@ -0,0 +1,235 @@
+			corev1.ResourceMemory: defaultVectorMemory,
+			corev1.ResourceCPU:    defaultVectorCpuRequest,
+		},
+	}
+	vectorContainer := NewContainer(constants.CollectorName, constants.VectorName, corev1.PullIfNotPresent, *resources)
+	// deliberately not passing any resources for running the below container process, let it have cpu and memory as the process requires
+	exporterresources := &corev1.ResourceRequirements{}
+
+	exporterContainer := NewContainer("logfilesmetricexporter", "logfilesmetricexporter", corev1.PullIfNotPresent, *exporterresources)
+
+	vectorContainer.Ports = []corev1.ContainerPort{
+		{
+			Name:          metricsPortName,
+			ContainerPort: metricsPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+
+	exporterContainer.Ports = []corev1.ContainerPort{
+		{
+			Name:          exporterPortName,
+			ContainerPort: exporterPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+
+	vectorContainer.Env = []corev1.EnvVar{
+		{Name: "LOG", Value: "debug"},
+		{Name: "VECTOR_SELF_NODE_NAME", ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "spec.nodeName"}}},
+		{Name: "METRICS_CERT", Value: "/etc/vector/metrics/tls.crt"},
+		{Name: "METRICS_KEY", Value: "/etc/vector/metrics/tls.key"},
+		{Name: "NODE_IPV4", ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "status.hostIP"}}},
+		{Name: "POD_IP", ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "status.podIP"}}},
+	}
+	proxyEnv := utils.GetProxyEnvVars()
+	vectorContainer.Env = append(vectorContainer.Env, proxyEnv...)
+
+	vectorContainer.VolumeMounts = []corev1.VolumeMount{
+		{Name: logContainers, ReadOnly: true, MountPath: logContainersValue},
+		{Name: logPods, ReadOnly: true, MountPath: logPodsValue},
+		{Name: logJournal, ReadOnly: true, MountPath: logJournalValue},
+		{Name: logAudit, ReadOnly: true, MountPath: logAuditValue},
+		{Name: logOvn, ReadOnly: true, MountPath: logOvnValue},
+		{Name: logOauthapiserver, ReadOnly: true, MountPath: logOauthapiserverValue},
+		{Name: logOpenshiftapiserver, ReadOnly: true, MountPath: logOpenshiftapiserverValue},
+		{Name: logKubeapiserver, ReadOnly: true, MountPath: logKubeapiserverValue},
+		{Name: config, ReadOnly: true, MountPath: vectorConfigValue},
+		{Name: dataDir, ReadOnly: false, MountPath: vectorDataDir},
+		{Name: localtime, ReadOnly: true, MountPath: localtimeValue},
+		{Name: tmp, MountPath: tmpValue},
+	}
+	exporterContainer.VolumeMounts = []corev1.VolumeMount{
+		{Name: logVolumeMountName, MountPath: logVolumePath},
+		{Name: metricsVolumeName, MountPath: metricsVolumeValue},
+	}
+	// Setting up CMD for log-file-metric-exporter
+	exporterContainer.Command = []string{"/usr/local/bin/log-file-metric-exporter", "  -verbosity=2", " -dir=/var/log/containers", " -http=:2112", " -keyFile=/etc/fluent/metrics/tls.key", " -crtFile=/etc/fluent/metrics/tls.crt"}
```

Comment:
I assume there is a place in future to extract this to common code for both collectors?